OpenRA.Game/Server/PlayerMessageTracker.cs

```diff
@@ -0,0 +1,78 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenRA.Server
+{
+	class PlayerMessageTracker
+	{
+		readonly Dictionary<int, List<long>> messageTracker = new Dictionary<int, List<long>>();
+		readonly Server server;
+		readonly Action<Connection, int, int, byte[]> dispatchOrdersToClient;
+		readonly Action<Connection, string, Dictionary<string, object>> sendLocalizedMessageTo;
+
+		public PlayerMessageTracker(Server server, Action<Connection, int, int, byte[]> dispatchOrdersToClient, Action<Connection, string, Dictionary<string, object>> sendLocalizedMessageTo)
+		{
+			this.server = server;
+			this.dispatchOrdersToClient = dispatchOrdersToClient;
+			this.sendLocalizedMessageTo = sendLocalizedMessageTo;
+		}
+
+		public void DisableChat(Connection conn, uint time)
+		{
+			dispatchOrdersToClient(conn, 0, 0, new Order("DisableChatEntry", null, false) { ExtraData = time }.Serialize());
+		}
+
+		public bool UpdatePlayerFloodLimits(Connection conn, string chatDisabled)
```

Comment:
This returns a bool, so it should have a name indicating what the bool means (the bool does not indicate whether updating succeeded). What about something like `IsPlayerAtFloodLimit` and then invert the return values?

Reply:
renamed

Reply:
You still need to swap the return values. Now `IsPlayerAtFloodLimit` returns `true` if they aren't.

Reply:
good catch

Reply:
The method name is misleading because it suggests that this is simply a check but in fact it does much more under the hood. But it simplifies things so I guess we can let it slide...